test/api/projects_api_test.rb

```diff
@@ -11,10 +11,105 @@ def app
   end
 
   def test_can_get_projects
-    user = FactoryBot.create(:user, :student, enrol_in: 0)
-    get with_auth_token('/api/projects', user)
+    # Create a new dummy unit
+    unit = FactoryBot.create(:unit)
+
+    # Get a new project from unit
+    new_project = unit.active_projects.first
+    
+    # A student enroll to new_project
+    student = new_project.student
+
+    # Perform Get   new_project.main_convenor_user
+    get with_auth_token "/api/projects/#{new_project.id}", new_project.student
+    
+    # Check if the call success
     assert_equal 200, last_response.status
+
+    # Check the returned details match as new_project
+    assert_equal new_project.id, last_response_body['project_id']
+
+    # check if the details posted match as new_project    
+    response_keys = %w(unit_id campus_id enrolled) 
+    project = Project.find(last_response_body['project_id'])
+    assert_json_matches_model(last_response_body, project, response_keys)
+
+    # Check if the details in the newly created project match as the new_project    
+    assert_equal project['unit_id'], new_project.unit_id
+    assert_equal project['campus_id'], new_project.campus_id
+    assert_equal project['enrolled'], new_project.enrolled
+  end 
+
+  # PUT test
+  def test_change_project_campus
+    project = Project.first
```

Comment:
Please use FactoryBot to create new unit, then call the project from that unit...